Makefile

```diff
@@ -47,12 +47,17 @@ precommit: dependabot-check license-check generate build lint test
 
 .PHONY: test-with-coverage
 test-with-coverage:
-	set -e; for dir in $(ALL_COVERAGE_MOD_DIRS); do \
+	set -e; \
+	printf "" > coverage.txt; \
+	for dir in $(ALL_COVERAGE_MOD_DIRS); do \
 	  echo "go test ./... + coverage in $${dir}"; \
 	  (cd "$${dir}" && \
 	    $(GOTEST_WITH_COVERAGE) ./... && \
-	    go tool cover -html=coverage.txt -o coverage.html); \
-	done
+	    go tool cover -html=coverage.out -o coverage.html); \
+	  [ -f "$${dir}/coverage.out" ] && cat "$${dir}/coverage.out" >> coverage.txt; \
+	done; \
+	sed -i.bak -e '2,$$ { /^mode: /d; }' coverage.txt; \
```

Comment:
Is the back up file needed here or can we remove that?

Reply:
Removed by having a null extension backup.  

Yes, there's the warning about how this will cause a corrupted file if the disk is full, but more likely our build will have other problems if something like that happens -- an invalid coverage file is probably the least of our worries.

Reply:
Hmm, looks like that caused this error:

```
sed: can't read : No such file or directory
```

This isn't a critical ask if you just want to revert back to the working state. Sorry to cause this issue.